kpack-image-builder/controllers/buildreconcilerinfo_controller.go

```diff
@@ -0,0 +1,170 @@
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"code.cloudfoundry.org/korifi/controllers/api/v1alpha1"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	BuildReconcilerInfoName = "kpack-image-builder"
+	ReadyConditionType      = "Ready"
+)
+
+func NewBuildReconcilerInfoReconciler(c client.Client, scheme *runtime.Scheme, log logr.Logger, clusterBuilderName, rootNamespaceName string) *BuildReconcilerInfoReconciler {
+	return &BuildReconcilerInfoReconciler{
+		Client:             c,
+		Scheme:             scheme,
+		Log:                log,
+		ClusterBuilderName: clusterBuilderName,
+		RootNamespaceName:  rootNamespaceName,
+	}
+}
+
+// BuildReconcilerInfoReconciler reconciles a BuildReconcilerInfo object
+type BuildReconcilerInfoReconciler struct {
+	client.Client
+	Scheme             *runtime.Scheme
+	Log                logr.Logger
+	ClusterBuilderName string
+	RootNamespaceName  string
+}
+
+//+kubebuilder:rbac:groups=korifi.cloudfoundry.org,resources=buildreconcilerinfos,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=korifi.cloudfoundry.org,resources=buildreconcilerinfos/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=korifi.cloudfoundry.org,resources=buildreconcilerinfos/finalizers,verbs=update
+
+//+kubebuilder:rbac:groups=kpack.io,resources=clusterbuilders,verbs=get;list;watch
+//+kubebuilder:rbac:groups=kpack.io,resources=clusterbuilders/status,verbs=get
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the BuildReconcilerInfo object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.0/pkg/reconcile
+func (r *BuildReconcilerInfoReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	if req.Name != BuildReconcilerInfoName || req.Namespace != r.RootNamespaceName {
+		return ctrl.Result{}, nil
+	}
+
+	info := new(v1alpha1.BuildReconcilerInfo)
+	err := r.Client.Get(ctx, req.NamespacedName, info)
+	if err != nil {
+		r.Log.Error(err, "Error when fetching BuildReconcilerInfo") // TODO: ensure we're using correct log format
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	clusterBuilder := new(buildv1alpha2.ClusterBuilder)
+	err = r.Client.Get(ctx, types.NamespacedName{Name: r.ClusterBuilderName}, clusterBuilder)
+	if err != nil {
+		r.Log.Error(err, "Error when fetching ClusterBuilder") // TODO: ensure we're using correct log format
```

Comment:
I'm not sure it necessarily makes sense to do it here, but mentioning this just to raise awareness. In some of the `CFTask` reconcilers we're now emitting K8s events to surface some of these warnings more visibly.

Example:
https://github.com/cloudfoundry/korifi/pull/1155/files#diff-58eb0aa24cc37d5046b0ce2c2ecebd69d2e006014778c2f72c3a263776ddd745R87

I think the status is fine for this singleton type of resource, but it could be interesting for some of the RunWorkload type work that's upcoming. 🤔

Reply:
Good to know. The BuildWorkload reconciler could probably benefit from events as well.